src/providers/ldap/sdap_async_nested_groups.c

```diff
@@ -1912,8 +1939,7 @@ sdap_nested_group_lookup_user_send(TALLOC_CTX *mem_ctx,
     /* search */
     subreq = sdap_get_generic_send(state, ev, group_ctx->opts, group_ctx->sh,
                                    member->dn, LDAP_SCOPE_BASE, filter, attrs,
-                                   group_ctx->opts->user_map,
-                                   group_ctx->opts->user_map_cnt,
+				   NULL, 0,
```

Comment:
Hi,

why is it needed to remove the map here?

I think this approach is good as well, but you are doing too many shortcuts with this patch.

Imo it would be better to handle the FSPs in `sdap_nested_group_lookup_unknown_send()/recv()` by adding a dedicated `sdap_nested_group_lookup_fsp_send()/recv()` request to handle FSPs in the same way as plain user and group members.

bye,
Sumit

Reply:
Hi,
We can't use maps for ldapsearch with multiple object classes in the filter due to the design limitation. I wanted to use single ldap search when detecting FSPs/users. To me it's unnecessary to introduce another `sdap_nested_group_lookup_fsp_send()/recv()` as it would mean another pointless ldapsearch. 
Since we only require member uid at this stage, there is no real need for maps and we can squeeze both searches into single one.
Ondrej

Reply:
Hi,
Well, yes, that would be an ideal case (as the current code is bit "cumbersome") yes. 
The problem is, that in order to search for all objectclasses we need to use unmapped search (I've mentioned this in a different PR).
Using unmapped search means `sdap_parse_entry()` will not map entries for us so we gotta do it manually. Now I did it for the user objectclass as it's not a big deal (single attr uid), but not sure what we need for group objectclasses.

Ondrej

Reply:
Hi,

you can just request all attributes and run `sdap_parse_entry()` after the type is known based on the objectclass. There is an example how it can be done in `sdap_asq_search_parse_entry()`.

bye,
Sumit

Reply:
I see, but `sdap_parse_entry()` needs `sdap_msg` parameter, where do I get it from `group_ctx`?

Reply:
Hi,

instead of using `sdap_get_generic_send()` you can use `sdap_get_generic_ext_send()` and add your own parser as a callback. This callback will receive `sdap_msg`.

HTH

bye,
Sumit

Reply:
Hi,
Sorry, that's bit too complex for me. Can you suggest a suitable code? I can't figure out myself.
Ideally I would like to parse attributes in `sdap_nested_group_lookup_unknown_recv()` if that's possible.
Namely I need better replacement of this snippet:
```
       ret = sysdb_attrs_get_string(*_entry, state->group_ctx->opts->user_map[SDAP_AT_USER_NAME].name, &val);
       if (ret != EOK) {
          DEBUG(SSSDBG_TRACE_ALL, "Unable to get username for %s\n",val);
          goto done;
       }
       /* we need to populate the sys_name in the user map so the user is recognized later on */
       sysdb_attrs_add_string(*_entry, state->group_ctx->opts->user_map[SDAP_AT_USER_NAME].sys_name, val);

```
Once I have the parsing done, then I could possibly do the rest myself.
Thanks.
Ondrej